modules/es6.promise.js

```diff
@@ -55,195 +45,66 @@ var useNative = function(){
-    aFunction(executor);
-    var record = {
-      p: strictNew(this, P, PROMISE),         // <- promise
-      c: [],                                  // <- awaiting reactions
-      a: undefined,                           // <- checked in isUnhandled reactions
-      s: 0,                                   // <- state
-      d: false,                               // <- done
-      v: undefined,                           // <- value
-      h: false,                               // <- handled rejection
-      n: false                                // <- notify
-    };
-    this[RECORD] = record;
-    try {
-      executor(ctx($resolve, record, 1), ctx($reject, record, 1));
-    } catch(err){
-      $reject.call(record, err);
+  P = Yaku;
+
+  // TODO: use core-js [[toString]]
+  // I haven't read much of the architecture of core-js, temporarily hard-coded it.
+  Yaku.toString = function () { return '[native code]'; };
+  Yaku.prototype.then.toString = function () { return '[native code]'; };
+  Yaku.prototype['catch'].toString = function () { return '[native code]'; };
+  Yaku.resolve.toString = function () { return '[native code]'; };
+  Yaku.reject.toString = function () { return '[native code]'; };
+  Yaku.all.toString = function () { return '[native code]'; };
+  Yaku.race.toString = function () { return '[native code]'; };
+  // TODO END
+
+  require('./$.mix')(P.prototype, Yaku.prototype);
+
+  P.nextTick = asap;
+  P.onUnhandledRejection = function (value, promise) {
+    var handler, console;
+    if(isNode){
+      process.emit('unhandledRejection', value, promise);
+    } else if(handler = global.onunhandledrejection){
+      handler({promise: promise, reason: value});
+    } else if((console = global.console) && console.error){
+      console.error('Unhandled promise rejection', value);
     }
   };
-  require('./$.mix')(P.prototype, {
-    // 25.4.5.3 Promise.prototype.then(onFulfilled, onRejected)
-    then: function then(onFulfilled, onRejected){
-      var S = anObject(anObject(this).constructor)[SPECIES];
-      var react = {
-        ok:   typeof onFulfilled == 'function' ? onFulfilled : true,
-        fail: typeof onRejected == 'function'  ? onRejected  : false
-      };
-      var promise = react.P = new (S != undefined ? S : P)(function(res, rej){
-        react.res = res;
-        react.rej = rej;
-      });
-      aFunction(react.res);
-      aFunction(react.rej);
-      var record = this[RECORD];
-      record.c.push(react);
-      if(record.a)record.a.push(react);
-      if(record.s)notify(record, false);
-      return promise;
-    },
-    // 25.4.5.1 Promise.prototype.catch(onRejected)
-    'catch': function(onRejected){
-      return this.then(undefined, onRejected);
-    }
-  });
 }
 
 // export
 $def($def.G + $def.W + $def.F * !useNative, {Promise: P});
 require('./$.tag')(P, PROMISE);
 species(P);
-species(Wrapper = require('./$.core')[PROMISE]);
+species(require('./$.core')[PROMISE]);
```

Comment:
W/o it `@@species` will not be added to wrapper of native constructor in `library` version.